### DISCUSSION TITLE
compute_msd for multiple groups

### DISCUSSION THREAD

**Discussion by EsmeeBerger** [brucefan1983/GPUMD#753] [Q&A]
I would like to perform a MSD calculation for multiple groups during the same run. I have two atom types in my system, and I have defined a grouping method with one group for each atom type.

I tried putting the `compute_msd` keyword twice in the run.in file, like this (of course with some values for the sample interval and Nc): 
`compute_msd <sample_interval> <Nc> group 0 0`
`compute_msd <sample_interval> <Nc> group 0 1`
but this only yields one msd.out file containing one MSD calculation. 

I have also tried giving two groups at once for a single `compute_msd` keyword, like this:
`compute_msd <sample_interval> <Nc> group 0 0 group 0 1`
but it resulted in a syntax error.

Is it possible to compute the MSD for multiple groups during one MD simulation?

**Comment by brucefan1983** [comment]
I know this feature is highly desired, but unfortunately it is not available yet. Currently the best way is to use two separate runs:

```
#equilibration at NVT/NPT
ensemble npt_xxx (or nvt_xxx)
run xxx

#calculate the MSD/SDC for one group (group_id_1)
ensemble nve
compute_msd <sample_interval> <Nc> group <group_method> <group_id_1>
run xxx

#calculate the MSD/SDC for the other group (group_id_2)
ensemble nve
compute_msd <sample_interval> <Nc> <group_method> <group_id_2>
run xxx

# even more runs if you need...

```

**Comment by EsmeeBerger** [comment]
I understand, thank you for the response!
